### PR TITLE
[docs] Update README.osx

### PR DESCRIPTION
**Pull request by fetzerch**
Explain how to use cmake to generate the xcode project files.

I've also changed the file to markdown. Is that something we want, or should I change it back to plain text? 

Preview:
- https://github.com/fetzerch/xbmc/blob/doc_osx/docs/README.osx.md
- https://github.com/fetzerch/xbmc/blob/doc_osx/docs/README.ios.md

Will update the others once we agreed on the format and style.

ping: @Memphiz, @hudokkow, @wsnipex

### CODE REVIEW

**Comment by Memphiz** [comment]
I like the markdown - well done with the adaptions imo - thx

**Comment by fetzerch** [comment]
Thanks. Ok osx, ios fully done.

@Memphiz can you check if the versions in the ios doc are ok? I added sierra and el capitan. From what I understood Xcode 8 is not supported until you PR goes in? Also I removed the line recommending ios sdk 8.1 because the sdk version should match the device people want to build for.

**Comment by fetzerch** [comment]
Any volunteers for one of the other files? :)

**Comment by hudokkow** [comment]
I can do linux/Ubuntu/RPi. Just not today probably.

**Comment by stefansaraev** [comment]
https://github.com/xbmc/xbmc/tree/master/Kodi.xcodeproj seems seems not to used now

**Comment by Memphiz** [comment]
@fetzerch nope the sdk advise needs to stay. There were issues when people used SDKs that were not the ones i developed for. For example the external screen behavior when using airplay mirroring behaves different when using different SDK versions regardless of the used deployment target (which stayed the same of course). At least that was the case around ios 8.x SDKs (might be no issue with newer SDKs anymore but i didn't check that). So while i never thought that would be the case - Apple developers screwed up at times ;).

**Comment by fetzerch** [comment]
@Memphiz Thanks for the explaination, added it back. Any other things you'd want me to fix? Otherwise we can get it in.

File: docs/README.ios.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+There are 3 ways to build Kodi for iOS:
+
+- Xcode IDE (easiest as it presents the build system in a GUI environment)
+- command-line with xcodebuild
+- command-line with make
+
+Kodi for iOS is composed of a main binary with numerous dynamic libraries and
+codecs that support a multitude of music and video formats.
+
+The minimum version of iOS you need to run(!) Kodi is 5.1.1 atm.
```
Comment (Memphiz) [inline docs/README.ios.md]: Could you correct this to 6.0 while you are at it?

File: docs/README.ios.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+The dependencies are built into `tools/depends` and installed into
+`/Users/Shared/xbmc-depends`.
+
+**ADVANCED developers only**: If you want to specify an iOS SDK version (if
+multiple versions are installed) - then append it to the configure line
+above. The example below would use the iOS SDK 8.0:
+
+    ./configure --host=arm-apple-darwin --with-sdk=8.0
+
+Ensure that you also adapt the Xcode project to use this SDK version.
```
Comment (Memphiz) [inline docs/README.ios.md]: This one needs to be adapted i guess

File: docs/README.ios.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+Before you can use Xode to build Kodi, the Xcode project has to be generated
+with CMake. Note that CMake is compiled as parts of the depends doesn't have
+to be installed separately. Also a Toolchain-file has been generated with is
+used to configure CMake.
+
+    mkdir $HOME/Kodi/build
+    cd $HOME/Kodi/build
+    /Users/Shared/xbmc-depends/buildtools-native/bin/cmake -G Xcode -DCMAKE_TOOLCHAIN_FILE=/Users/Shared/xbmc-depends/iphoneos9.3_armv7-target/share/Toolchain.cmake ../project/cmake
+
+The toolchain file location differs depending on your iOS and SDK version.
```
Comment (Memphiz) [inline docs/README.ios.md]: please make it clear that the iphoneos9.3_armv7-target has to be replaced with iphoneos<SDKVERSION>_<CPU>-target ... i fear we get support questions here if its not in baby speech ;).

File: docs/README.ios.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    ./configure --host=arm-apple-darwin --with-sdk=8.0
+
+Ensure that you also adapt the Xcode project to use this SDK version.
+
+### 3.3 Compile Kodi binary addons
+
+Kodi maintains a set of binary addons (PVR clients, Visualizations, Audio DSP
+plugins and more). They can be built as shown below:
+
+    make -C target/binary-addons
```
Comment (Memphiz) [inline docs/README.ios.md]: this is out of context now - please ensure that people are in tools/depends before trying to issue this command.

File: docs/README.ios.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+**NOTE**: If you only want to build specific addons you can specify like this:
+
+    make -C target/binary-addons ADDONS="pvr.hts pvr.dvblink"
```
Comment (Memphiz) [inline docs/README.ios.md]: same here

File: docs/README.ios.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+Be sure to select a device configuration. Building for simulator is not
+supported.
+
+The build process will take a long time when building the first time.
+You can see the progress in "Build Results". There are a large number of static
+and dynamic libraries that will need to be built. Once these are built,
+subsequent builds will be faster.
+
+After the build, you can ether run Kodi for iOS from Xcode.
```
Comment (Memphiz) [inline docs/README.ios.md]: ether might be either, but "either" without "or" is strange too :)

File: docs/README.ios.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+Alternatively, you can also build via Xcode from the command-line with
+xcodebuild, triggered by CMake:
+
+    cd $HOME/Kodi/build
+    cmake --build . --config "Debug" -- -verbose -jobs $(sysctl -n hw.ncpu)
```
Comment (Memphiz) [inline docs/README.ios.md]: we could use getconf _NPROCESSORS_ONLN here aswell then right?

File: docs/README.osx.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+5 Xcode 6.4.0 against OSX SDK 10.10 (Y)
+6 Xcode 7.x against OSX SDK 10.11 (EC)
+7 Xcode 8.0 against OSX SDK 10.12 (EC)
+8 Xcode 8.0 against OSX SDK 10.12 (S)
+
+### 3.2 Install Kodi build depends
+
+Kodi requires a set of build dependencies to be built and installed before you
+will be able to build the Kodi main binary. These often just called *depends*
+are installed using the commands described below (with the latest iOS SDK found
```
Comment (Memphiz) [inline docs/README.osx.md]: iOS SDK should be MacOS SDK

File: docs/README.osx.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+The dependencies are built into `tools/depends` and installed into
+`/Users/Shared/xbmc-depends`.
+
+**ADVANCED developers only**: If you want to specify an macOS SDK version (if
+multiple versions are installed) - then append it to the configure line
+above. The example below would use the macOS SDK 10.9:
+
+    ./configure --host=arm-apple-darwin --with-sdk=10.9
+
+Ensure that you also adapt the Xcode project to use this SDK version.
```
Comment (Memphiz) [inline docs/README.osx.md]: This needs to be adapted in whatever way this needs to be done now with cmake generation...

File: docs/README.osx.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+### 3.3 Compile Kodi binary addons
+
+Kodi maintains a set of binary addons (PVR clients, Visualizations, Audio DSP
+plugins and more). They can be built as shown below:
+
+    make -C target/binary-addons
```
Comment (Memphiz) [inline docs/README.osx.md]: same out of context problem - we need to ensure users are in tools/depends before trying this ...

File: docs/README.osx.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+
+**NOTE**: If you only want to build specific addons you can specify like this:
+
+    make -C target/binary-addons ADDONS="pvr.hts pvr.dvblink"
```
Comment (Memphiz) [inline docs/README.osx.md]: same here

File: docs/README.osx.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+Before you can use Xode to build Kodi, the Xcode project has to be generated
+with CMake. Note that CMake is compiled as parts of the depends doesn't have
+to be installed separately. Also a Toolchain-file has been generated with is
+used to configure CMake.
+
+    mkdir $HOME/Kodi/build
+    cd $HOME/Kodi/build
+    /Users/Shared/xbmc-depends/buildtools-native/bin/cmake -G Xcode -DCMAKE_TOOLCHAIN_FILE=/Users/Shared/xbmc-depends/macosx10.12_x86_64-target/share/Toolchain.cmake ../project/cmake
+
+The toolchain file location differs depending on your macOS and SDK version.
```
Comment (Memphiz) [inline docs/README.osx.md]: spoonfeed what needs to be changed here.

File: docs/README.osx.md [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+
+Or the release version:
+
+    $HOME/Kodi/build/Release/kodi.bin
+
+Alternatively, you can also build via Xcode from the command-line with
+xcodebuild, triggered by CMake:
+
+    cd $HOME/Kodi/build
+    cmake --build . --config "Debug" -- -verbose -jobs $(sysctl -n hw.ncpu)
```
Comment (Memphiz) [inline docs/README.osx.md]: getconf _NPROCESSORS_ONLN

**Review by Memphiz — CHANGES_REQUESTED** [review]
Looks good beside my remarks - thx alot for this. I think the next PR would be removing the Xcode project as it has no value now anymore.

**Comment by fetzerch** [comment]
Thanks, I've tried to fix all comments, but i'm not sure with the `--with-sdk`. Need to check if the sdk version is put into the toolchain file and picked up from there automatically or if there's further work needed.

Dropping the xcode project is done in: https://github.com/xbmc/xbmc/pull/11217

**Review by Memphiz — APPROVED** [review]

**Comment by fetzerch** [comment]
@Memphiz: I've installed the 10.9 sdk now from https://github.com/phracker/MacOSX-SDKs and `--with-sdk` seems to do the trick. cmake picks it up from the toolchain file. Unfortunately it compiles only when generating a make file project. With xcodebuild and xcode i get this. Any idea?

```
=== BUILD TARGET XBMCHelper OF PROJECT kodi WITH CONFIGURATION Release ===

Check dependencies

Ld /Users/jcf/develop/kodi/kodi-build-xcode-old/tools/darwin/runtime/XBMCHelper normal x86_64
    cd /Users/jcf/develop/kodi/kodi
    export MACOSX_DEPLOYMENT_TARGET=10.9
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -L/Users/jcf/develop/kodi/kodi-build-xcode-old/tools/darwin/runtime -F/Users/jcf/develop/kodi/kodi-build-xcode-old/tools/darwin/runtime -filelist /Users/jcf/develop/kodi/kodi-build-xcode-old/build/XBMCHelper/kodi.build/Release/XBMCHelper.build/Objects-normal/x86_64/XBMCHelper.LinkFileList -mmacosx-version-min=10.9 -Xlinker -object_path_lto -Xlinker /Users/jcf/develop/kodi/kodi-build-xcode-old/build/XBMCHelper/kodi.build/Release/XBMCHelper.build/Objects-normal/x86_64/XBMCHelper_lto.o -fobjc-link-runtime -L/Users/Shared/xbmc-depends/macosx10.9_x86_64-target/lib -Wl,-search_paths_first -arch x86_64 -mmacosx-version-min=10.8 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -stdlib=libc++ -Wl,-headerpad_max_install_names -framework IOKit -framework Carbon -framework Cocoa -Xlinker -dependency_info -Xlinker /Users/jcf/develop/kodi/kodi-build-xcode-old/build/XBMCHelper/kodi.build/Release/XBMCHelper.build/Objects-normal/x86_64/XBMCHelper_dependency_info.dat -o /Users/jcf/develop/kodi/kodi-build-xcode-old/tools/darwin/runtime/XBMCHelper
Undefined symbols for architecture x86_64:
  "_objc_readClassPair", referenced from:
      __ARCLite__load() in libarclite_macosx.a(arclite.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
```

**Comment by Memphiz** [comment]
No idea - but what is strange ist that deployment target is set to 10.8 in cflags (which is correct) but to 10.9 via export MACOSX_DEPLOYMENT_TARGET (which is wrong). 
Also this error is for XBMCHelper right? I think it has a static xcode project still no? ARC is some sort of automated garbage collection afair...

**Comment by fetzerch** [comment]
Thanks, sorted it out. Now it's really just building depends with the `--with-sdk` parameter and using the respective toolchain file.

btw: any chance of rebasing https://github.com/xbmc/xbmc/pull/10539 now that krypton is branched?

**Pull request merged**